commit 7e6739b9336e61fe23ca4e2c8d1fda8f19f979bf
Merge: a47e60729d96 65898687cf73
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Wed Oct 5 11:24:12 2022 -0700

    Merge tag 'drm-next-2022-10-05' of git://anongit.freedesktop.org/drm/drm
    
    Pull drm updates from Dave Airlie:
     "Lots of stuff all over, some new AMD IP support and gang submit
      support. i915 has further DG2 and Meteorlake pieces, and a bunch of
      i915 display refactoring. msm has a shrinker rework. There are also a
      bunch of conversions to use kunit.
    
      This has two external pieces, some MEI changes needed for future Intel
      discrete GPUs. These should be acked by Greg. There is also a cross
      maintainer shared tree with some backlight rework from Hans in here.
    
      Core:
       - convert selftests to kunit
       - managed init for more objects
       - move to idr_init_base
       - rename fb and gem cma helpers to dma
       - hide unregistered connectors from getconnector ioctl
       - DSC passthrough aux support
       - backlight handling improvements
       - add dma_resv_assert_held to vmap/vunmap
    
      edid:
       - move luminance calculation to core
    
      fbdev:
       - fix aperture helper usage
    
      fourcc:
       - add more format helpers
       - add DRM_FORMAT_Cxx, DRM_FORMAT_Rxx, DRM_FORMAT_Dxx
       - add packed AYUV8888, XYUV8888
       - add some kunit tests
    
      ttm:
       - allow bos without backing store
       - rewrite placement to use intersect/compatible functions
    
      dma-buf:
       - docs update
       - improve signalling when debugging
    
      udmabuf:
       - fix failure path GPF
    
      dp:
       - drop dp/mst legacy code
       - atomic mst state support
       - audio infoframe packing
    
      panel:
       - Samsung LTL101AL01
       - B120XAN01.0
       - R140NWF5 RH
       - DMT028VGHMCMI-1A T
       - AUO B133UAN02.1
       - IVO M133NW4J-R3
       - Innolux N120ACA-EA1
    
      amdgpu:
       - Gang submit support
       - Mode2 reset for RDNA2
       - New IP support:
            DCN 3.1.4, 3.2
            SMU 13.x
            NBIO 7.7
            GC 11.x
            PSP 13.x
            SDMA 6.x
            GMC 11.x
       - DSC passthrough support
       - PSP fixes for TA support
       - vangogh GFXOFF stats
       - clang fixes
       - gang submit CS cleanup prep work
       - fix VRAM eviction issues
    
      amdkfd:
       - GC 10.3 IP ISA fixes
       - fix CRIU regression
       - CPU fault on COW mapping fixes
    
      i915:
       - align fw versioning with kernel practices
       - add display substruct to i915 private
       - add initial runtime info to driver info
       - split out HDCP and backlight registers
       - MEI XeHP SDV GSC support
       - add per-gt sysfs defaults
       - TLB invalidation improvements
       - Disable PCI BAR resize on 32-bit
       - GuC firmware updates and compat changes
       - GuC log timestamp translation
       - DG2 preemption workaround changes
       - DG2 improved HDMI pixel clocks support
       - PCI BAR sanity checks
       - Enable DC5 on DG2
       - DG2 DMC fw bumped
       - ADL-S PCI ID added
       - Meteorlake enablement
       - Rename ggtt_view to gtt_view
       - host RPS fixes
       - release mmaps on rpm suspend on discrete
       - clocking and dpll refactoring
       - VBT definitions and parsing updates
       - SKL watermark code extracted to separate file
       - allow seamless M/N changes on eDP panels
       - BUG_ON removal and cleanups
    
      msm:
       - DPU:
           simplified VBIF configuration
           cleanup CTL interfaces
       - DSI:
           removed unused msm_display_dsc_config struct
           switch regulator calls to new API
           switched to PANEL_BRIDGE for direct attached panels
       - DSI_PHY: convert drivers to parent_hws
       - DP: cleanup pixel_rate handling
       - HDMI: turned hdmi-phy-8996 into OF clk provider
       - misc dt-bindings fixes
       - choose eDP as primary display if it's available
       - support getting interconnects from either the mdss or the mdp5/dpu
         device nodes
       - gem: Shrinker + LRU re-work:
       - adds a shared GEM LRU+shrinker helper and moves msm over to that
       - reduce lock contention between retire and submit by avoiding the
         need to acquire obj lock in retire path (and instead using resv
         seeing obj's busyness in the shrinker
       - fix reclaim vs submit issues
       - GEM fault injection for triggering userspace error paths
       - Map/unmap optimization
       - Improved robustness for a6xx GPU recovery
    
      virtio:
       - improve error and edge conditions handling
       - convert to use managed helpers
       - stop exposing LINEAR modifier
    
      mgag200:
       - split modeset handling per model
    
      udl:
       - suspend/disconnect handling improvements
    
      vc4:
       - rework HDMI power up
       - depend on PM
       - better unplugging support
    
      ast:
       - resolution handling improvements
    
      ingenic:
       - add JZ4760(B) support
       - avoid a modeset when sharpness property is unchanged
       - use the new PM ops
    
      it6505:
       - power seq and clock updates
    
      ssd130x:
       - regmap bulk write
       - use atomic helpers instead of simple helpers
    
      via:
       - rename via_drv to via_dri1, consolidate all code.
    
      radeon:
       - drop DP MST experimental support
       - delayed work flush fix
       - use time_after
    
      ti-sn65dsi86:
       - DP support
    
      mediatek:
       - MT8195 DP support
       - drop of_gpio header
       - remove unneeded result
       - small DP code improvements
    
      vkms:
       - RGB565, XRGB64 and ARGB64 support
    
      sun4i:
       - tv: convert to atomic
    
      rcar-du:
       - Synopsys DW HDMI bridge DT bindings update
    
      exynos:
       - use drm_display_info.is_hdmi
       - correct return of mixer_mode_valid and hdmi_mode_valid
    
      omap:
       - refcounting fix
    
      rockchip:
       - RK3568 support
       - RK3399 gamma support"
    
    * tag 'drm-next-2022-10-05' of git://anongit.freedesktop.org/drm/drm: (1374 commits)
      drm/amdkfd: Fix UBSAN shift-out-of-bounds warning
      drm/amdkfd: Track unified memory when switching xnack mode
      drm/amdgpu: Enable sram on vcn_4_0_2
      drm/amdgpu: Enable VCN DPG for GC11_0_1
      drm/msm: Fix build break with recent mm tree
      drm/panel: simple: Use dev_err_probe() to simplify code
      drm/panel: panel-edp: Use dev_err_probe() to simplify code
      drm/panel: simple: Add Multi-Inno Technology MI0800FT-9
      dt-bindings: display: simple: Add Multi-Inno Technology MI0800FT-9 panel
      drm/amdgpu: correct the memcpy size for ip discovery firmware
      drm/amdgpu: Skip put_reset_domain if it doesn't exist
      drm/amdgpu: remove switch from amdgpu_gmc_noretry_set
      drm/amdgpu: Fix mc_umc_status used uninitialized warning
      drm/amd/display: Prevent OTG shutdown during PSR SU
      drm/amdgpu: add page retirement handling for CPU RAS
      drm/amdgpu: use RAS error address convert api in mca notifier
      drm/amdgpu: support to convert dedicated umc mca address
      drm/amdgpu: export umc error address convert interface
      drm/amdgpu: fix sdma v4 init microcode error
      drm/amd/display: fix array-bounds error in dc_stream_remove_writeback()
      ...

diff --cc drivers/gpu/drm/amd/amdgpu/mes_v11_0.c
index f92744b8d79d,b48684db2832..5cec6b259b7f
--- a/drivers/gpu/drm/amd/amdgpu/mes_v11_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/mes_v11_0.c
@@@ -183,12 -185,16 +185,21 @@@ static int mes_v11_0_add_hw_queue(struc
  	mes_add_queue_pkt.trap_handler_addr = input->tba_addr;
  	mes_add_queue_pkt.tma_addr = input->tma_addr;
  	mes_add_queue_pkt.is_kfd_process = input->is_kfd_process;
 +	mes_add_queue_pkt.trap_en = 1;
 +
 +	/* For KFD, gds_size is re-used for queue size (needed in MES for AQL queues) */
 +	mes_add_queue_pkt.is_aql_queue = input->is_aql_queue;
 +	mes_add_queue_pkt.gds_size = input->queue_size;
  
+ 	if (!(((adev->mes.sched_version & AMDGPU_MES_VERSION_MASK) >= 4) &&
+ 		  (adev->ip_versions[GC_HWIP][0] >= IP_VERSION(11, 0, 0)) &&
+ 		  (adev->ip_versions[GC_HWIP][0] <= IP_VERSION(11, 0, 3))))
+ 		mes_add_queue_pkt.trap_en = 1;
+ 
+ 	/* For KFD, gds_size is re-used for queue size (needed in MES for AQL queues) */
+ 	mes_add_queue_pkt.is_aql_queue = input->is_aql_queue;
+ 	mes_add_queue_pkt.gds_size = input->queue_size;
+ 
  	return mes_v11_0_submit_pkt_and_poll_completion(mes,
  			&mes_add_queue_pkt, sizeof(mes_add_queue_pkt),
  			offsetof(union MESAPI__ADD_QUEUE, api_status));